### PR TITLE
Improves the code & the sprite of my donator item

### PR DESCRIPTION
**Pull request by GoldenAlpharex**
## About The Pull Request

Before you lose your mind, no, it doesn't make it a free pair of welding goggles. It just makes it so you can slap them with welding goggles to give them welding protection, on a separate toggle (this took much more code than expected).

There's also two non-modular changes that are just stuff I'm going to PR upstream today, and I know they'll get merged because I'm the reason those issues were introduced. Now the code for the `alternative_worn_layer` will be better.
I also edited the sprite a bit to make it fit the intended hairstyle that they're going to be worn over.

They also play a cute little noise when the welding shutters are toggled.

## How This Contributes To The Skyrat Roleplay Experience
Those goggles are something very cute and flavorful that I had to live without for 9 months. I am now reunited with my steampunk goggles :)

## Changelog

No changelog for this, as I'm the only one that will see any change from this.

### CODE REVIEW

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm [hunk 1]
```diff
@@ -985,10 +985,129 @@
...
+
+/obj/item/clothing/glasses/welding/goldengoggles/weldingvisortoggle(mob/user)
+	. = ..()
+	handle_sight_updating(user)
+
+/obj/item/clothing/glasses/welding/goldengoggles/attackby(obj/item/attacking_item, mob/living/user, params)
+	if(!istype(attacking_item, /obj/item/clothing/glasses/welding))
+		..()
+	if(welding_upgraded)
+		to_chat(user, span_warning("\The [src] was already upgraded to have welding protection!"))
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm:1038]:
```suggestion
		to_chat(user, span_warning("\The [src] were already upgraded to have welding protection!"))
```

could do `[src.p_were()]` but that feels like overkill

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm [hunk 2]
```diff
@@ -985,10 +985,129 @@
...
+/obj/item/clothing/glasses/welding/goldengoggles/proc/toggle_shutters(mob/user)
+	if(!can_use(user) || !user)
+		return FALSE
+	if(!toggle_welding_protection(user))
+		return FALSE
+
+	to_chat(user, span_notice("You slide \the [src]'s welding shutters slider, [welding_protection ? "closing" : "opening"] them."))
+	playsound(user, shutters_sound, 100, TRUE)
+	if(iscarbon(user))
+		var/mob/living/carbon/C = user
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]:
```suggestion
		var/mob/living/carbon/carbon_user = user
```
I'm disappointed.

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm [hunk 3]
```diff
@@ -985,10 +985,129 @@
...
+		C.head_update(src, forced = 1)
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]:
```suggestion
		carbon_user.head_update(src, forced = 1)
```

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm (hunk 1 above)
Comment (GoldenAlpharex) [reply modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm:1038]: Yeah, I don't plan on having them change quantity ;)

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm (hunk 2 above)
Comment (GoldenAlpharex) [reply modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm]: You know, I copy-pasted that and never noticed.

File: modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm (hunk 1 above)
Comment (tf-4) [reply modular_skyrat/modules/customization/modules/clothing/~donator/donator_clothing.dm:1038]: You never know with subtypes, but yes it's unlikely

**Comment by thestubborn** [comment]
i dont have any code words for this but these are very cute I like them, congrats on the cool item

**Comment by GoldenAlpharex** [comment]
> i dont have any code words for this but these are very cute I like them, congrats on the cool item

Thanks! It's Useroth's work, I just tweaked slightly a  thing or two, here and there.

**Comment by Useroth** [comment]
That's a lot of code, though, donator items being what they are actually warrant being somewhat unique.
Seems fine to me.

**Pull request merged**